### DISCUSSION TITLE
Theming v18

### DISCUSSION THREAD

**Discussion by Leonardonline** [primefaces/community#2759] [PrimeNG]
Hi everyone,

I'm looking for a space to discuss some challenges with the new theming system, design tokens, and related topics.

Would this forum be the right place, or would the Discord channel be more appropriate or other?

**Comment by qwadrox** [comment]
> Hi everyone,
> 
> I'm looking for a space to discuss some challenges with the new theming system, design tokens, and related topics.
> 
> Would this forum be the right place, or would the Discord channel be more appropriate or other?

I think wherever you feel more comfortable.

**Comment by nawlbergs** [comment]
I'm brand-new to primeNG.

I cannot seem to customize the color/token for my success/danger/info buttons: `success.color | --p-button-success-color`.
I would like to change the success color across entire app...

Where do you change the token they provide?  Im able to change the `semantic.primary` colors.... but i cannot get anything else to change.

```js
const myPreset = definePreset(Aura, {
    semantic: {
      primary:  palette('#336699'),
      success: {
        color: '#FF3388', 
        background: '#FF3388'   // NOPE...
      },
    },
    components: {
      button: {
        success: {
          color: '#FF3388',
          background: '#FF3388', // NOPE..
        },
      }
    }
});
usePreset(myPreset);
```

**Reply by qwadrox** [reply]
Hi @nawlbergs 

Here is what you should do to actually change the style: 
```ts
 components: {
    button: {
      colorScheme: {
        light: {
          root: {
            success: {
              background: '#FF3388',
            },
          },
        },
        dark: {
          root: {
            success: {
              background: '#FF3388',
            },
          },
        },
      },
    },
  },
```

<img width="573" alt="image" src="https://github.com/user-attachments/assets/da01b2a5-e8ea-4592-a5f9-01d46e886235">

**Reply by qwadrox** [reply]
@nawlbergs What you did would only work on the component level: 

```html
<p-button label="Success2" severity="success" [dt]="successStyle" />
```

```ts
successStyle = {
    success: {
      color: '#FF3388',
      background: '#FF3388', // NOPE..
    },
  };
```


Check out these YouTube videos. Although they're PrimeVue specific, but they might help anyway: 

https://www.youtube.com/watch?v=By04epRCpDA
https://www.youtube.com/watch?v=J3KFw5sih98

**Reply by nawlbergs** [reply]
Thanks I’ll try it tonight. 

Looks like I was missing the ‘root’ part. I tried about 20 variants trying to get it to work.

**Reply by qwadrox** [reply]
> Thanks I’ll try it tonight.
> 
> Looks like I was missing the ‘root’ part. I tried about 20 variants trying to get it to work.

You can console log the preset. It's an object, and you can compare it with yours.

**Reply by nawlbergs** [reply]
Thanks... console logging is quite helpful.

**Comment by Leonardonline** [comment]
The component  `p-tree` have several modifications in their child elements and classes, but neither the documentation nor the migration guide mention this.

**Reply by qwadrox** [reply]
PrimeNg 18 is still beta so don't expect to be perfect. I think they know there are lots of mistakes in the documentation and components as well and they're working on it. Once 18 arrives they're going to focus on maintenance, and if it's still not working we can open an issue.

**Reply by Leonardonline** [reply]
My goal, if it wasn't clear, was precisely to point out this issue so that it can be resolved or properly documented.
